### PR TITLE
Unread ticket status per user.

### PR DESCRIPTION
**Pull request by git-jls**
This pull request will make unread messages appear as grey inside the inbox.

This should satisfy the feature request: _Unread ticket status per user._

I included a .gif showing of the behavior for several different situations, including ticket creation from the non-signed in users ticket creation form (web gui) and tickets created through the MTA.

output_rv46j5

### CODE REVIEW

File: app/models/ticket.rb [hunk 1]
```diff
@@ -124,6 +126,14 @@ def set_default_notifications!
     self.notified_user_ids = users.map(&:id)
   end
 
+  def is_unread?(user)
+    unread_users.include? user
```
Comment (cthiel) [inline app/models/ticket.rb]: While this feature is pretty nice in principle, this change is making Brimir very slow to render for us (20+ seconds) on a system with some 100 open tickets and roughly 2000-3000 users, as this is triggering n+ slow queries per ticket when displaying them.

File: app/mailers/ticket_mailer.rb [hunk 2]
```diff
@@ -124,7 +124,8 @@ def receive(email)
         message_id: email.message_id,
         content_type: content_type,
         to_email_address: to_email_address,
-        raw_message: StringIO.new(email.to_s)
+        raw_message: StringIO.new(email.to_s),
+        unread_users: User.all # none of the users could have read this
```
Comment (cthiel) [inline app/mailers/ticket_mailer.rb]: We tried a temp workaround to set this to ```User.where(agent: true)```

Comment (cthiel) [inline app/mailers/ticket_mailer.rb]: And ran a dirty clean-up hack on the console:

```
class TicketsUser < ApplicationRecord
end

TicketsUser.count # before: 2077962
TicketsUser.delete_all(user_id: User.where(agent: false).pluck(:id))
TicketsUser.count # after: 22683
```

Now load times are again < 1s :)

Comment (frenkel) [inline app/mailers/ticket_mailer.rb]: Looks good, could you create a PR for this, then we'll include that change as well.

Comment (frenkel) [inline app/mailers/ticket_mailer.rb]: No forget I asked, I think this should be done together with a change in the other file. I'll try to come up with a better solution today.

Comment (cthiel) [inline app/mailers/ticket_mailer.rb]: That would be great! Thanks in advance.

Comment (frenkel) [inline app/mailers/ticket_mailer.rb]: It seems we can't do better than what I pushed to master a few seconds ago.

**Pull request merged**